packages/core/src/lib/TwaManifest.ts

```diff
@@ -294,15 +294,30 @@ export class TwaManifest {
       shortcuts: shortcuts,
       webManifestUrl: webManifestUrl.toString(),
       features: {},
-      // We're leaving `shareTarget` as undefined for now, as the shareTarget is not complete
-      // and we want to avoid creating manifests with invalid values for now. The broader Web Share
-      // Target implementation is being tracked at
-      // https://github.com/GoogleChromeLabs/bubblewrap/issues/21.
-      shareTarget: undefined,
+      shareTarget: TwaManifest.verifyShareTarget(webManifestUrl, webManifest.share_target),
     });
     return twaManifest;
   }
 
+  private static verifyShareTarget(
+      webManifestUrl: URL, shareTarget?: ShareTarget): ShareTarget | undefined {
+    if (!shareTarget || !shareTarget.action || !shareTarget.params || !shareTarget.params.files) {
```

Comment:
Can this not be `if (!shareTarget?.action || !shareTarget.params?.files)`?

Reply:
Good point. done.